frontend/src/utilities/quickStartUtils.ts

```diff
@@ -58,28 +68,37 @@ export const getQuickStartLabel = (
   return `${launchStatus}`;
 };
 
-export const isQuickStartInProgress = (
+export const getQuickStartCompletionStatus = (
   quickStartId?: string | null,
   qsContext?: QuickStartContextValues,
-): boolean => {
+): CompletionStatusEnum | undefined => {
   if (!quickStartId || !qsContext || !qsContext.allQuickStartStates) {
-    return false;
+    return undefined;
+  }
+  const quickStartState = qsContext.allQuickStartStates[quickStartId];
+  if (!quickStartState || quickStartState.taskNumber === -1) {
+    return undefined;
   }
-  const launchStatus = getLaunchStatus(quickStartId, qsContext);
-
-  return launchStatus === LaunchStatusEnum.Continue;
-};
 
-export const isQuickStartComplete = (
-  quickStartId?: string | null,
-  qsContext?: QuickStartContextValues,
-): boolean => {
-  if (!quickStartId || !qsContext || !qsContext.allQuickStartStates) {
-    return false;
+  if (quickStartState.status === QuickStartStatus.COMPLETE) {
+    let qsStatus = CompletionStatusEnum.Success,
+      currentStep = 0;
+    while (quickStartState[`taskStatus${currentStep}`]) {
+      const status = quickStartState[`taskStatus${currentStep}`];
+      if (status) {
+        if (status === QuickStartTaskStatus.FAILED) {
+          qsStatus = CompletionStatusEnum.Failed;
+        }
+        if (status === QuickStartTaskStatus.REVIEW || status === QuickStartTaskStatus.VISITED) {
+          return CompletionStatusEnum.InProgress;
```

Comment:
I see the expected result in the JIRA issue said: the QS card has the status "Failed" when at least one of the steps has been set as "failed". So I guess maybe we should better not return `InProgress` here? Instead, maybe it's better to return `Failed` above and only set `InProgress` here. What do you think? @dlabaj

Reply:
Originally had that but after speaking with @kywalker-rh unless the completed all the steps we should mark at as in progress.

Reply:
Makes sense

Reply:
Yes, just confirming we changed it slightly because In progress makes more sense for the user that hasn't completed every step.